### PR TITLE
Being able to set environment variables using 'envFrom'

### PR DESCRIPTION
**Pull request by SmaineTF1**
## What was changed
Add the ability to define environment variables that come from configmaps or secrets

## Why?
Make use of envFrom is pretty handy when, for instance you have all the environment variables defined in Secrets or Configmaps.
That way we only need to have this bit of code if we have all the config env vars in a secret

```yaml
web:
  additionalEnvSecret: 
    - "temporal-web-envvars"
```

### CODE REVIEW

**Review by mindaugasrukas — APPROVED** [review]
LGTM

**Review by underrun — APPROVED** [review]

File: templates/server-deployment.yaml [hunk 1]
```diff
@@ -121,6 +121,17 @@ spec:
           {{- if or $.Values.server.additionalEnv $serviceValues.additionalEnv }}
           {{- toYaml (default $.Values.server.additionalEnv $serviceValues.additionalEnv) | nindent 12 }}
           {{- end }}
+          {{- if or $serviceValues.additionalEnvCM $serviceValues.additionalEnvSecret }}
+          envFrom:
+            {{- if $serviceValues.additionalEnvCM }}
```
Comment (XaF) [inline templates/server-deployment.yaml:126]: Any reason not to handle those as lists? (vs. scalar)

Comment (SmaineTF1) [reply templates/server-deployment.yaml:126]: I thought that having a unique secret per component would be enough. I can work on handling a list of CM/Secrets.

Comment (SmaineTF1) [reply templates/server-deployment.yaml:140]: I just updated in order to take lists? @XaF

Comment (XaF) [reply templates/server-deployment.yaml:140]: Thanks!

**Review by XaF — APPROVED** [review]

**Comment by SmaineTF1** [comment]
What about this PR? Do you need anything else?

**Comment by sgrzemski** [comment]
+1 to that. 
@SmaineTF1 could you please fix the semgrep, so that @XaF @underrun or @mindaugasrukas can merge it?

**Comment by SmaineTF1** [comment]
@sgrzemski sorry, I'm confused but what should I do for that check? This is not obvious to me :/

**Comment by sgrzemski** [comment]
@SmaineTF1 :
image
Sorry, I thought it was in a failed state, but it's just queued to be triggerred manually.

File: values.yaml [hunk 2]
```diff
@@ -84,6 +84,8 @@ server:
   additionalVolumes: []
   additionalVolumeMounts: []
   additionalEnv: []
+  additionalEnvCM: []
```
Comment (robholland) [inline values.yaml:87]: Please use `additionalEnvConfigMap` rather than using a contraction.

**Review by robholland — CHANGES_REQUESTED** [review]
Thanks for the submission, happy to work with you on getting this merged. Please can you rebase this against master.

**Comment by robholland** [comment]
I'm going to close this PR due to inactivity. Please re-open if you'd like to continue the process.

**Pull request closed without merging**